osu.Game.Modes.Taiko/Beatmaps/TaikoBeatmapConverter.cs

```diff
@@ -41,6 +42,7 @@ public Beatmap<TaikoHitObject> Convert(Beatmap original)
         {
             return new Beatmap<TaikoHitObject>(original)
             {
+                TimingInfo = new LegacyTimingInfo(original.TimingInfo),

```

Comment:
I want this to construct LegacyTimingInfo only if this is a legacy beatmap. Consider the case of "new" beatmaps where `Convert` is passed in a Beatmap that contains base, interfaced, taiko hit objects, where legacy_velocity_multiplier will not be needed..

Similar to how this is done on line 55 of this file.

Best way to go about this in general is to make a `class LegacyControlPoint : ControlPoint`, with an internal ctor, then instantiate control points of that type from OsuLegacyDecoder.